composer/trainer/trainer.py

```diff
@@ -2982,7 +3008,7 @@ def _eval_loop(
                                     rank_num_samples -= 1
                                     num_samples_in_microbatch = data_spec.get_num_samples_in_batch(self.state.batch)
                                     # Skip updating metric if batch is only padded samples
-                                    if num_samples_in_microbatch == 1:
+                                    if num_samples_in_microbatch == 1 or hasattr(data_spec, 'spg_ws'):
```

Comment:
Why do we need to do this?

Reply:
For sequence parallelism, we only support microbatch size` 1/seq_parallel_ws`, that is, only one sample per microbatch, which gets distributed over the sequence parallel world. Hence, in that case we cannot reduce the microbatch size. That is why I set `skip_metric_update = True` for that case.

Reply:
What happens if I run sequence parallel + dtms 2? Wouldn't that still work?

Reply:
dtms has to be 1 for sequence parallelism. The reason being that if you can set dtms = 2, then you can also just halve the sequence parallel world size and set dtms = 1.

Reply:
Can we enforce this in validation?

Reply:
Added checks in composer to raise error if dtms > 1 for seq parallelism.